test-node.bash

```diff
@@ -70,28 +80,68 @@ while [[ $# -gt 0 ]]; do
             ;;
         --init-force)
             force_init=true
+            build_contracts=true
+            build_node_images=true
             shift
             ;;
         --dev)
             simple=false
             shift
             if [[ $# -eq 0 || $1 == -* ]]; then
                 # If no argument after --dev, set both flags to true
-                dev_build_nitro=true
-                dev_build_blockscout=true
+                dev_nitro=true
+                build_dev_nitro=true
+                dev_blockscout=true
+                build_dev_blockscout=true
             else
                 while [[ $# -gt 0 && $1 != -* ]]; do
                     if [[ $1 == "nitro" ]]; then
-                        dev_build_nitro=true
+                        dev_nitro=true
+                        build_dev_nitro=true
                     elif [[ $1 == "blockscout" ]]; then
-                        dev_build_blockscout=true
+                        dev_blockscout=true
+                        build_dev_blockscout=true
                     fi
                     shift
                 done
             fi
             ;;
         --build)
-            force_build=true
+            build_dev_nitro=true
+            build_dev_blockscout=true
+            build_contracts=true
+            build_node_images=true
+            shift
+            ;;
+        --no-build)
+            build_dev_nitro=false
+            build_dev_blockscout=false
+            build_contracts=false
+            build_node_images=false
+            shift
+            ;;
+        --build-dev-nitro)
+            build_dev_nitro=true
+            shift
+            ;;
+        --no-build-dev-nitro)
+            build_dev_nitro=false
+            shift
+            ;;
+        --build-dev-blockscout)
+            build_dev_blockscout=true
+            shift
+            ;;
+        --no-build-dev-blockscout)
+            build_dev_blockscout=false
+            shift
+            ;;
+        --build-contracts)
```

Comment:
I like the solution.

Small comment:
This isn't just contracts, but also scripts and tokenbridge. Can we rename it? build-iternal? build-utils? something similar..

Complicating things:
contracts and tokenbridge are weird dockers. They both fetch external code and build it according to NITRO_CONTRACTS_BRANCH, TOKEN_BRIDGE_BRANCH env variables. It means that it's very possible that rebuild is necessary even though docker doesn't recognize it. So we need an option to force rebuilding them. (docker build --force? I'm not sure how to do it)

I think a manual --build-contracts/internal/other should **force** rebuilding, but --init and --init-force should just do a normal build.
So if someone wants to force rebuild they'll do --init --build-contracts

Reply:
the option name could have force inside, like --force-build-utils or something

Reply:
Took your suggestions. The way to force rebuild with docker is `--no-cache`